libraries/AP_Filesystem/AP_Filesystem_ROMFS.cpp

```diff
@@ -40,17 +51,13 @@ int AP_Filesystem_ROMFS::open(const char *fname, int flags, bool allow_absolute_
     }
     if (idx == max_open_file) {
         errno = ENFILE;
+        AP_ROMFS::free(data);
         return -1;
     }
-    if (file[idx].data != nullptr) {
-        errno = EBUSY;
-        return -1;
-    }
```

Comment:
Isn't this check still just as necessary now as before?

Reply:
No, we know `file[idx].data == nullptr` from the loop above. We now have the lock to protect it from somebody else setting it `!= nullptr`.

This actually might have saved us (i.e. created a more noisy error) if it had been placed after the decompression but  the lock now ensures it can't occur.

Reply:
Ah, that makes sense, gotcha.